### ISSUE TITLE
Low MAP when training out custom dataset

### ISSUE THREAD

**Issue by genghuan2005** [MultimediaTechLab/YOLO#194]
## Describe the bug
A clear and concise description of what the bug is. Include any error messages or logs if available.

## System Info (please complete the following ## information):
 - OS: [e.g. Ubuntu 20.04]
 - Python Version: [e.g. 3.8]
 - PyTorch Version: [e.g. 1.8.1]
 - CUDA/cuDNN/MPS Version: [e.g. CUDA 11.1]
 - YOLO Model Version: [e.g. YOLOv9-c]

**Comment by Alan01252** [comment]
I think the problem is that the shuffle flag got lost to the DataLoader in this commit

https://github.com/MultimediaTechLab/YOLO/commit/95520e951eaea6ca51cda9fb468be91b1f0ea3af

Meaning the training data doesn't get shuffled properly during the training step.

**Comment by faridht** [comment]
I don't think this answers your question with respect to using .txt labels, but I have found that the .json format (i.e. COCO 1.0 format) worked well with me.

Also in my case I found two issues with the default settings:
- learning rate was too high from the start. I needed to reduce it (This was mentioned on a different issue here, but can't remember which one)
- Mosaic augmentation ruined my training accuracy because the objects I wanted to detect were already very small

Just my two cents based on my experience
